lib/rex/parser/arguments.rb

```diff
@@ -154,6 +158,29 @@ def short_arg_from_long_arg(long_arg)
 
         fmt_option.each { |opt| return opt if opt =~ SHORT_FLAG }
       end
+
+      # Parsing takes into account longest `short` flag found in 'arg' should as extracted first
+      #
+      # Returns Array of short arguments found in `arg`
+      def short_args_from_token(arg)
+        compare_arg = arg.dup[1..-1]
+        short_args = []
+        found_args = {}
+        fmt.keys.each do |keys|
```

Comment:
Note for future travelers: We should really document what the heck `fmt` is since even the variable name isn't really descriptive of what it contains, however for those curious it contains the list of command arguments that a given `msfconsole` command such as `sessions` supports:

```
[11] pry(#<Rex::Parser::Arguments>)> fmt
=> {["-c", "--command"]=>
  [true, "Run a command on the session given with -i, or all", "<command>"],
 ["-C", "--meterpreter-command"]=>
  [true, "Run a Meterpreter Command on the session given with -i, or all", "<command>"],
 ["-h", "--help"]=>[false, "Help banner"],
 ["-i", "--interact"]=>[true, "Interact with the supplied session ID", "<id>"],
 ["-l", "--list"]=>[false, "List all active sessions"],
 ["-v", "--list-verbose"]=>[false, "List all active sessions in verbose mode"],
 ["-d", "--list-inactive"]=>[false, "List all inactive sessions"],
 ["-q", "--quiet"]=>[false, "Quiet mode"],
 ["-k", "--kill"]=>[true, "Terminate sessions by session ID and/or range", "<id>"],
 ["-K", "--kill-all"]=>[false, "Terminate all sessions"],
 ["-s", "--script"]=>
  [true, "Run a script or module on the session given with -i, or all", "<script>"],
 ["-u", "--upgrade"]=>
  [true, "Upgrade a shell to a meterpreter session on many platforms", "<id>"],
 ["-t", "--timeout"]=>[true, "Set a response timeout (default: 15)", "<seconds>"],
 ["-S", "--search"]=>[true, "Row search filter.", "<filter>"],
 ["-x", "--list-extended"]=>[false, "Show extended information in the session table"],
 ["-n", "--name"]=>[true, "Name or rename a session by ID", "<id> <name>"]}
[12] pry(#<Rex::Parser::Arguments>)> 
```

Happy to add some documentation for this but seems like atm the documentation is listed as `:nodoc:`.

Reply:
Oh and `keys.each` will bring up the key list however its a little odd here since the keys is actually an array here so you will get an array of arrays:

```
[12] pry(#<Rex::Parser::Arguments>)> fmt.keys
=> [["-c", "--command"],
 ["-C", "--meterpreter-command"],
 ["-h", "--help"],
 ["-i", "--interact"],
 ["-l", "--list"],
 ["-v", "--list-verbose"],
 ["-d", "--list-inactive"],
 ["-q", "--quiet"],
 ["-k", "--kill"],
 ["-K", "--kill-all"],
 ["-s", "--script"],
 ["-u", "--upgrade"],
 ["-t", "--timeout"],
 ["-S", "--search"],
 ["-x", "--list-extended"],
 ["-n", "--name"]]
[13] pry(#<Rex::Parser::Arguments>)> 
```